sympy/vector/vector.py

```diff
@@ -250,6 +250,30 @@ def outer(self, other):
 
         return DyadicAdd(*args)
 
+    def vec_project(self, other, scalar=False):
+        """
+        Returns the vector or scalar projection of the 'other' on 'self'.
+
+        Examples
+        ========
+
+        >>> from sympy.vector.coordsysrect import CoordSysCartesian
+        >>> from sympy.vector.vector import Vector, BaseVector
+        >>> C = CoordSysCartesian('C')
+        >>> i, j, k = C.base_vectors()
+        >>> v1 = i + j + k
+        >>> v2 = 3*i + 4*j
+        >>> v1.vec_project(v2)
+        7/3*C.i + 7/3*C.j + 7/3*C.k
+        >>> v1.vec_project(v2, scalar=True)
+        7/3
+
+        """
```

Comment:
@srjoglekar246 , You are right. If  'self' is a zero vector then the angle becomes undefined. In this case, the projection should return zero. I have added that in the most recent commit/